rubicon/objc/api.py

```diff
@@ -368,26 +368,36 @@ class MySubclass(NSObject):
     the generated setter keeps the stored object retained, and releases it when it is replaced.
 
     In a custom Objective-C protocol, only the metadata for the property is generated.
+
+    If ``weak`` is ``True``, the property will be created as a weak property. When assigning an object to it,
+    the reference count of the object will not be increased. When the object is deallocated, the property
+    value is set to None.
     """
 
-    def __init__(self, vartype=objc_id):
+    def __init__(self, vartype=objc_id, weak=False):
         super().__init__()
 
         self.vartype = ctype_for_type(vartype)
+        self.weak = weak
 
     def _get_property_attributes(self):
         attrs = [
             objc_property_attribute_t(b'T', encoding_for_ctype(self.vartype)),  # Type: vartype
         ]
         if issubclass(self.vartype, objc_id):
-            attrs.append(objc_property_attribute_t(b'&', b''))  # retain
+            reference = b'W' if self.weak else b'&'
+            attrs.append(objc_property_attribute_t(reference, b''))
         return (objc_property_attribute_t * len(attrs))(*attrs)
 
     def class_register(self, class_ptr, attr_name):
-        add_ivar(class_ptr, '_' + attr_name, self.vartype)
+
+        ivar_name = '_' + attr_name
```

Comment:
It would be an attribute which is not set during `__init__` but only after `class_register` or `protocol_register` is called, since the `objc_property` does not have a "name" independent of the class / protocol that it's registered with. So we'd only save a single call of composing the `ivar_name` in the `delloc_callback`. Not sure if it's worth it...

Reply:
Ah, you're right - I overlooked that `attr_name` is a parameter of both methods and so can't be set in `__init__`. Ignore my comment then 🙂